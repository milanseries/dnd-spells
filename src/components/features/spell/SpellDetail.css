/* Spell-detailDetails.css */

.spell-detail-card {
  min-width: 300px !important;
  background-color: #f1f1f1 !important;
  border: 1px solid #ddd !important;
  border-radius: 8px !important;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1) !important;
  padding: 16px !important;
}

.spell-detail-title {
  color: #333 !important;
  font-size: 24px !important;
  font-weight: bold !important;
  margin-bottom: 8px !important;
}

.spell-detail-level {
  color: #666 !important;
}

.spell-detail-desc {
  color: #444 !important;
  margin-bottom: 16px !important;
}

.spell-detail-list {
  color: #666 !important;
  margin-bottom: 16px !important;
}

.spell-detail-higher-level {
  color: #666 !important;
  margin-bottom: 16px !important;
}

.spell-detail-damage {
  color: #ff5722 !important;
  margin-bottom: 8px !important;
}
