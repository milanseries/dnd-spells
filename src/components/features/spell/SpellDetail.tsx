/* eslint-disable @typescript-eslint/restrict-template-expressions */
/* eslint-disable @typescript-eslint/naming-convention */
import {
  Card,
  CardContent,
  Typography,
  List,
  ListItem,
  ListItemText,
} from '@mui/material'

import './SpellDetail.css'

const SpellDetail = ({ spell }: any) => {
  const {
    name,
    desc,
    range,
    components,
    material,
    duration,
    level,
    damage,
    school,
    higher_level,
    casting_time,
  } = spell

  return (
    <Card variant="outlined" className="spell-card">
      <CardContent>
        <Typography variant="h5" component="div" className="spell-title">
          {name}
        </Typography>
        <Typography color="text.secondary" gutterBottom className="spell-level">
          Level {level} {school.name} spell
        </Typography>
        <Typography variant="body2" className="spell-desc">
          {desc[0]}
        </Typography>

        <List dense className="spell-list">
          <ListItem>
            <ListItemText primary="Range" secondary={range} />
          </ListItem>
          <ListItem>
            <ListItemText
              primary="Components"
              secondary={components.join(', ')}
            />
          </ListItem>
          <ListItem>
            <ListItemText primary="Material" secondary={material} />
          </ListItem>
          <ListItem>
            <ListItemText primary="Duration" secondary={duration} />
          </ListItem>
          <ListItem>
            <ListItemText primary="Casting Time" secondary={casting_time} />
          </ListItem>
        </List>

        {higher_level && (
          <>
            <Typography variant="h6" component="div" className="spell-title">
              At Higher Levels
            </Typography>
            <Typography variant="body2" className="spell-higher-level">
              {higher_level[0]}
            </Typography>
          </>
        )}

        {damage && (
          <>
            <Typography variant="h6" component="div" className="spell-title">
              Damage
            </Typography>
            {Object.keys(
              damage.damage_at_slot_level || damage.damage_at_character_level
            ).map((level) => (
              <Typography key={level} variant="body2" className="spell-damage">
                {`Level ${level}: ${
                  damage.damage_at_slot_level?.[level] ??
                  damage.damage_at_character_level?.[level]
                } ${damage.damage_type.name} damage`}
              </Typography>
            ))}
          </>
        )}
      </CardContent>
    </Card>
  )
}

export default SpellDetail
