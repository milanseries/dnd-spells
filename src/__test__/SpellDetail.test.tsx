import React from 'react'
import { render } from '@testing-library/react'
import SpellDetail from '../components/features/spell/SpellDetail'
import { mockSpell } from './mocks/data.mock'

describe('SpellDetail component', () => {
  test('renders correctly', () => {
    const { asFragment } = render(<SpellDetail spell={mockSpell} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
