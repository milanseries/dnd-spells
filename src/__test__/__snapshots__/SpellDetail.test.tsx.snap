// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpellDetail component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root spell-detail-card css-1v5z4dq-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
          >
            Acid Arrow
          </h4>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            A shimmering green arrow streaks toward a target within range and bursts in a spray of acid. Make a ranged spell attack against the target. On a hit, the target takes 4d4 acid damage immediately and 2d4 acid damage at the end of its next turn. On a miss, the arrow splashes the target with acid for half as much of the initial damage and no damage at the end of its next turn.
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >
            Damage
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Level 2 Damage: 4d4
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Level 3 Damage: 5d4
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Level 4 Damage: 6d4
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Level 5 Damage: 7d4
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Level 6 Damage: 8d4
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Level 7 Damage: 9d4
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Level 8 Damage: 10d4
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Level 9 Damage: 11d4
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >
            Higher Levels
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            When you cast this spell using a spell slot of 3rd level or higher, the damage (both initial and later) increases by 1d4 for each slot level above 2nd.
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Range: 90 feet
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Components: V, S, M
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Material: Powdered rhubarb leaf and an adder's stomach.
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Duration: Instantaneous
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Casting Time: 1 action
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
